slick/src/main/scala/slick/ast/Comprehension.scala

```diff
@@ -9,9 +9,11 @@ final case class Comprehension(sym: TermSymbol, from: Node, select: Node, where:
                                groupBy: Option[Node] = None, orderBy: ConstArray[(Node, Ordering)] = ConstArray.empty,
                                having: Option[Node] = None,
                                distinct: Option[Node] = None,
-                               fetch: Option[Node] = None, offset: Option[Node] = None) extends DefNode {
+                               fetch: Option[Node] = None,
+                               offset: Option[Node] = None,
+                               forUpdate: Option[Node] = None) extends DefNode {
```

Comment:
I'd use a Boolean for now until it's clear how to really handle the more complex case. It's probably similar to how `distinct` and `distinctOn` are implemented. I wouldn't be surprised if the most sensible definition of `forUpdate` also turns out to be `forUpdateOn(identity)`. This would cause a change in semantics from the current version (not a reason to hold back on this feature, just something to watch out for).